Dockerfile

```diff
@@ -136,209 +61,321 @@ RUN chmod 755 /*.sh /usr/local/bin/* \
+        espeak=1.48.04+dfsg-7+deb10u1 \
+        ffmpeg=7:4.1.6-1~deb10u1 \
+        lame=3.100-2+b1 \
+        libnmap-parser-perl=1.37-1 \
+        libttspico-utils=1.0+git20130326-9 \
+        mp3wrap=0.5-4 \
+        mpg123=1.25.10-2 \
+        mplayer=2:1.3.0-8+b4 \
+        nmap=7.70+dfsg1-6+deb10u1 \
+        normalize-audio=0.7.7-15 \
+        snmp=5.7.3+dfsg-5+deb10u1 \
+        snmp-mibs-downloader=1.2 \
+        sox=14.4.2+git20190427-1 \
+        vorbis-tools=1.4.0-11 \
       && LC_ALL=C apt-get autoremove -qqy && LC_ALL=C apt-get clean \
       && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* ~/.[^.] ~/.??* ~/* \
     ; fi
 
-# Add Perl basic app layer for pre-compiled packages
-RUN LC_ALL=C DEBIAN_FRONTEND=noninteractive apt-get update \
-    && LC_ALL=C DEBIAN_FRONTEND=noninteractive apt-get install -qqy --no-install-recommends \
-        perl-base \
-        libarchive-extract-perl \
-        libarchive-zip-perl \
-        libcgi-pm-perl \
-        libcpanel-json-xs-perl \
-        libdbd-mariadb-perl \
-        libdbd-mysql \
-        libdbd-mysql-perl \
-        libdbd-pg-perl \
-        libdbd-pgsql \
-        libdbd-sqlite3 \
-        libdbd-sqlite3-perl \
-        libdbi-perl \
-        libdevice-serialport-perl \
-        libdevice-usb-perl \
-        libgd-graph-perl \
-        libgd-text-perl \
-        libimage-imlib2-perl \
-        libimage-info-perl \
-        libimage-librsvg-perl \
-        libio-all-perl \
-        libio-file-withpath-perl \
-        libio-interface-perl \
-        libio-socket-inet6-perl \
-        libio-socket-ssl-perl \
-        libjson-perl \
-        libjson-pp-perl \
-        libjson-xs-perl \
-        liblist-moreutils-perl \
-        libmail-gnupg-perl \
-        libmail-imapclient-perl \
-        libmail-sendmail-perl \
-        libmime-base64-perl \
-        libmime-lite-perl \
-        libnet-server-perl \
-        libsocket6-perl \
-        libterm-readline-perl-perl \
-        libtext-csv-perl \
-        libtext-diff-perl \
-        libtext-iconv-perl \
-        libtimedate-perl \
-        libutf8-all-perl \
-        libwww-curl-perl \
-        libwww-perl \
-        libxml-libxml-perl \
-        libxml-parser-lite-perl \
-        libxml-parser-perl \
-        libxml-simple-perl \
-        libxml-stream-perl \
-        libxml-treebuilder-perl \
-        libxml-xpath-perl \
-        libxml-xpathengine-perl \
-        libyaml-libyaml-perl \
-        libyaml-perl \
-    && LC_ALL=C apt-get autoremove -qqy && LC_ALL=C apt-get clean \
-    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* ~/.[^.] ~/.??* ~/*
+
+# Custom image layers options:
+ARG IMAGE_LAYER_PERL_EXT="1"
 
 # Add Perl extended app layer for pre-compiled packages
 RUN if [ "${IMAGE_LAYER_PERL_EXT}" != "0" ]; then \
       LC_ALL=C DEBIAN_FRONTEND=noninteractive apt-get update \
       && LC_ALL=C DEBIAN_FRONTEND=noninteractive apt-get install -qqy --no-install-recommends \
-          perl \
```

Comment:
again, no pinning here. the intention is to rely on the latest version of the upstream environment.
FHEM Installer and Updater Modules would otherwise also complain about existing updates. Does not make sense to always be behind with updates even when pulling the latest image.

Reply:
The pinning is needed, to invalidate the cache if there is a new version. Otherwise if the layer doesn't change, the cache is never invalidated. 

I'm aware, that there needs to be some update jobs done. Currently i'm hoping that this will get into dependabot soon, so every time a new package is releases, a new build can be triggered to be up to date.
With the old behaivour, i think, there was no information available what versions are installed, and if somebody builds it on his own, he will get different results for the same dockerfile and revision, if the build job is run on a different time.